From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: nds-ports <nds-ports@localhost>
Date: Wed, 30 Sep 2026 13:41:47 +0200
Subject: [PATCH] Keep savegames per game in saves/<iwad>/

Savegames go to saves/<iwad>/doomsavN.dsg in the config directory, so
different games don't overwrite each other's slots.
---
 doomgeneric/m_config.c | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/doomgeneric/m_config.c b/doomgeneric/m_config.c
index 865e168..588b64d 100644
--- a/doomgeneric/m_config.c
+++ b/doomgeneric/m_config.c
@@ -2115,7 +2115,14 @@ char *M_GetSaveGameDir(char *iwadname)
 
         free(topdir);
 #else
-        savegamedir = M_StringJoin(configdir, DIR_SEPARATOR_S, ".savegame/", NULL);
+        // Separate savegames per IWAD, eg. ./saves/doom2.wad/doomsav0.dsg
+
+        char *topdir = M_StringJoin(configdir, DIR_SEPARATOR_S, "saves", NULL);
+
+        M_MakeDirectory(topdir);
+        savegamedir = M_StringJoin(topdir, DIR_SEPARATOR_S, iwadname,
+                                   DIR_SEPARATOR_S, NULL);
+        free(topdir);
 
         M_MakeDirectory(savegamedir);
 
